### PR TITLE
Add Java variants to Messages examples.

### PR DESCRIPTION
**Pull request by Peppe**
- Remove Message example
- Add Java variations of each example
- Better example data

### CODE REVIEW

**Comment by Peppe** [comment]
I'm having problems developing this locally, so I thought of doing a draft PR and verifying that it works similarly on Netlify as locally.

File: src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                "Linsey Listy", person.getPictureUrl());
+        message2.setUserAbbreviation("LL");
+        message2.setUserColorIndex(2);
+        list.setItems(Arrays.asList(message1, message2));
+
+        VerticalLayout chatLayout = new VerticalLayout(list, input);
+        chatLayout.setHeight("500px");
+        chatLayout.setWidth("400px");
+        chatLayout.expand(list);
+        // end::snippet[]
```
Comment (yuriy-fix) [inline src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java:54]: Missing `add(chatLayout)` is a part of the problem not displaying the example

Comment (Peppe) [reply src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java:55]: Done.

File: articles/ds/components/messages/index.asciidoc [hunk 2]
```diff
@@ -58,8 +51,22 @@ userColorIndex: number;
 userImg: string;
 ----
 
-Here is an example of an initialized `items` array with two messages:
+Here is an example of an initialized `items` array with two messages, in Java and TypeScript:
```
Comment (pekam) [inline articles/ds/components/messages/index.asciidoc:54]: I think that we shouldn't have source code as part of the text like this. We have the code examples for that purpose, and with those you only see them in your preferred language.

Looking at the other component demos, they mostly speak on a more general level about the component in the text part, and then let the code examples show the API.

File: src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+public class MessagesBasic extends Div {
+
+    public MessagesBasic() {
+        // tag::snippet[]
+        MessageList list = new MessageList();
+        MessageInput input = new MessageInput();
+        input.addSubmitListener(submitEvent -> {
+            MessageListItem newMessage = new MessageListItem(
+                    submitEvent.getValue(), Instant.now(), "Milla Sting");
+            newMessage.setUserAbbreviation("MS");
```
Comment (pekam) [inline src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java:30]: I don't think we need to use `setUserAbbreviation` just for the sake of covering all the APIs. It's just extra line of code which doesn't have any effect.

File: articles/ds/components/messages/index.asciidoc [hunk 4]
```diff
@@ -13,23 +13,16 @@ page-links:
 Messages is a collection of components to display messages and to build chats, comment sections and activity logs.
```
Comment (pekam) [inline articles/ds/components/messages/index.asciidoc:13]: A bit outside the scope of this PR, but I would avoid saying that you can build activity logs until you actually can.
```suggestion
Messages is a collection of components to display messages and to build chats and comment sections.
```

**Review by pekam — CHANGES_REQUESTED** [review]

File: articles/ds/components/messages/index.asciidoc (hunk 2 above)
Comment (Peppe) [reply articles/ds/components/messages/index.asciidoc:54]: Is your suggestion to delete the the code snippets? 

I added it after in DX tests, people was looking on how to define `items`, and didn't know what all was possible to define in these objects (especially in JS). But yeah, the practice is not there in other components.

File: src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java (hunk 3 above)
Comment (Peppe) [reply src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java:30]: Okay. We should take it away from the TS version as well to have them match.

File: frontend/demo/component/messages/MessageListItem.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
 export default interface MessageListItem {
```
[the remark this answers is not in the document]
Comment (Peppe) [reply frontend/demo/component/messages/MessageListItem.ts]: No we don't need it anymore, and it isn't used. Deleted.

File: articles/ds/components/messages/index.asciidoc (hunk 2 above)
Comment (pekam) [reply articles/ds/components/messages/index.asciidoc:54]: Maybe the TS code example should be changed from:
```html
<vaadin-message-list .items=${this.items}></vaadin-message-list>
```
to:
```html
<vaadin-message-list .items=${[
  {
    text: "foo"
  }..,
]}></vaadin-message-list>
```

File: src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java (hunk 3 above)
Comment (Peppe) [reply src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java:30]: Done.

**Comment by jouni** [comment]
Build failure:

```
11:55:34 AM: [ERROR] ERROR in /opt/build/repo/frontend/demo/component/messages/message-basic.ts
11:55:34 AM: [ERROR] [tsl] ERROR in /opt/build/repo/frontend/demo/component/messages/message-basic.ts(8,29)
11:55:34 AM: [ERROR]       TS2307: Cannot find module './MessageListItem' or its corresponding type declarations.
11:55:34 AM: [ERROR]
```

File: articles/ds/components/messages/index.asciidoc (hunk 2 above)
Comment (Peppe) [reply articles/ds/components/messages/index.asciidoc]: I can see the point and I changed it. 

However, I'm not that much of a fan myself as inline defining messages is not what I would do in any real-world application and thus I don't consider this best practice. But I'm fine with it.

Done.

File: src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (pekam) [inline src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java:30]:
```suggestion
```

File: src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                "Nature does not hurry, yet everything gets accomplished.",
+                LocalDateTime.now().minusDays(1).toInstant(ZoneOffset.UTC),
+                "Matt Mambo");
+        message1.setUserColorIndex(1);
+        MessageListItem message2 = new MessageListItem(
+                "Using your talent, hobby or profession in a way that makes you contribute with something good to this world is truly the way to go.",
+                LocalDateTime.now().minusMinutes(55).toInstant(ZoneOffset.UTC),
+                "Linsey Listy", person.getPictureUrl());
+        message2.setUserColorIndex(2);
+        list.setItems(Arrays.asList(message1, message2));
```
Comment (pekam) [inline src/main/java/com/vaadin/demo/component/messages/MessagesBasic.java:48]:
```suggestion
        list.setItems(message1, message2);
```

File: articles/ds/components/messages/index.asciidoc (hunk 2 above)
Comment (pekam) [reply articles/ds/components/messages/index.asciidoc]: Yeah, following best practices is of course a good idea, but it is just more important to easily see from the example what the property looks like.

And in any case, we're hard-coding the array in the code, so it's kinda impossible to really follow best practices. :sweat_smile:

**Review by pekam — APPROVED** [review]
These message demos are still standing out from other components as they go into API details in the descriptions, but changing those more is outside the scope of this PR.

**Pull request merged**